### PR TITLE
[Metricbeat] Add googlecloud compute overview dashboard

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
This PR is to add googlecloud compute overview dashboard:
Screen Shot 2020-03-17 at 10 33 08 AM (2)

closes https://github.com/elastic/beats/issues/16534

### CODE REVIEW

File: x-pack/metricbeat/module/googlecloud/_meta/docs.asciidoc [hunk 1]
```diff
@@ -59,3 +59,21 @@ A recommended `period` value between fetches is between 5 and 10 minutes, depend
...
+[float]
+=== `compute`
+Compute Metricset to fetch metrics from https://cloud.google.com/compute/[Compute Engine]
+Virtual Machines in Google Cloud Platform. The `compute` Metricset contains some of the
+metrics exported from the https://cloud.google.com/monitoring/api/metrics_gcp#gcp-compute[Stackdriver API].
+Extra labels and metadata are also extracted using the https://cloud.google.com/compute/docs/reference/rest/v1/instances/get[Compute API].
+This is enough to get most of the info associated with a metric like compute
+labels and metadata and metric specific Labels.
+
+The compute metricset comes with a predefined dashboard:
```
Comment (mtojek) [inline x-pack/metricbeat/module/googlecloud/_meta/docs.asciidoc]: nit: `compute`

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -210,6 +210,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Add `redisenterprise` module. {pull}16482[16482] {issue}15269[15269]
 - Align fields to ECS and add more tests for the azure module. {issue}16024[16024] {pull}16754[16754]
 - Add PubSub metricset to Google Cloud Platform module {pull}15536[15536]
+- Add overview dashboard for googlecloud compute. {issue}16534[16534] {pull}16819[16819]
```
Comment (mtojek) [inline CHANGELOG.next.asciidoc]: I think both changelog entries should be consistent :)

File: x-pack/metricbeat/module/googlecloud/_meta/docs.asciidoc [hunk 3]
```diff
@@ -59,3 +59,21 @@ A recommended `period` value between fetches is between 5 and 10 minutes, depend
...
+
+[float]
+== Metricsets
+Currently, we have `compute`, `loadbalancing`, and `pubsub` metricset in
+`googlecloud` module.
+
+[float]
+=== `compute`
+Compute Metricset to fetch metrics from https://cloud.google.com/compute/[Compute Engine]
+Virtual Machines in Google Cloud Platform. The `compute` Metricset contains some of the
```
Comment (mtojek) [inline x-pack/metricbeat/module/googlecloud/_meta/docs.asciidoc]: nit: metricset contains  (lowercase)

File: CHANGELOG.next.asciidoc (hunk 2 above)
Comment (kaiyan-sheng) [reply CHANGELOG.next.asciidoc]: 😂 I prefer googlecloud module but I don't want to modify the previous changelog entry 😬

**Comment by kaiyan-sheng** [comment]
> * The filters can be combined into one control instead of having 3 separate ones.

Good point, done!

> * `instance.uptime` can be added to the dashboard as a label between the filters and CPU graph.

I just added uptime as a separate visualization. The problem is instance/uptime metric from GCP is delta kind, which means it measures the change in a value over a sample period. So with this uptime visualization, you won't be able to see numbers going up to larger than 60.

**Comment by sorantis** [comment]
> The problem is instance/uptime metric from GCP is delta kind, which means it measures the change in a value over a sample period. So with this uptime visualization, you won't be able to see numbers going up to larger than 60.

@kaiyan-sheng I see the issue. I was thinking about it as a True/False indicator(sorry for not making it clear), e.g. `<green>Running 34<green>`, `<red>Down 19</red>`. If there's no other way of interpreting this metric, then we should probably not show it at all.

**Comment by kaiyan-sheng** [comment]
@sorantis Thank you for the suggestion. I changed uptime to show <green>100% Running 34<green> and <yellow> less than 100% Running 2<yellow>. With how googlecloud metrics are collected, this is still not _accurate_ . Because when you set period=5m in gcp compute metricset, you get metrics every minute still for 3 times in each 5 minute. 
Screen Shot 2020-03-09 at 1 12 13 PM 
I will put this PR into work in progress till I figure out how to accurately display this uptime. Thanks!

**Comment by kaiyan-sheng** [comment]
@sorantis What do you think about changing uptime using a gauge to show percentage here?  I'd like to show True/False indicator but the data we are collecting here doesn't give me a good way of displaying the exact number of instances. I'm also working on figuring out how we can improve data collection shown as screenshot above. After figuring that out, I can come back and revisit the uptime visualization. Thanks!

**Comment by sorantis** [comment]
looks much better, @kaiyan-sheng, thanks for revisiting it!

**Comment by sorantis** [comment]
nit: perhaps we can align all dropdowns vertically, and place the gauge between the filters and cpu utilization? this should save space, improve symmetry (and help my OCD :D)

**Comment by kaiyan-sheng** [comment]
@sorantis Totally understand. I just made the change. Thanks!

**Pull request merged**